codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/transformers/EventStreamNormalizer.kt

```diff
@@ -20,13 +26,34 @@ import software.amazon.smithy.rust.codegen.util.isEventStream
  */
 object EventStreamNormalizer {
     fun transform(model: Model): Model = ModelTransformer.create().mapShapes(model) { shape ->
-        if (shape is UnionShape && shape.isEventStream()) {
+        if (shape is OperationShape && shape.isEventStream(model)) {
+            addStreamErrorsToOperationErrors(model, shape)
+        } else if (shape is UnionShape && shape.isEventStream()) {
             syntheticEquivalent(model, shape)
         } else {
             shape
         }
     }
 
+    private fun addStreamErrorsToOperationErrors(model: Model, operation: OperationShape): OperationShape {
```

Comment:
1. True, thanks. Fixed it.
2. Also true, that it indicates fault, but I made it so that the client writes `yield Err( ClientEventStreamError::... )` and similarly for the server; and so the client `match (_) => Err( ServerEventStreamError::... )`. As above, I'm happy to change the names however you think they make more sense. As for "If an error is included in an input stream, then it is a client error. If it is included in an output stream, then it is a server error. If it is included in an input stream and an output stream, then it is both" I haven't seen this in the docs, I'm asking the team to confirm this